packages/actions/log/src/modules/log/sagas.js

```diff
@@ -0,0 +1,44 @@
+import {all, call, put, takeEvery} from 'redux-saga/effects'
+import {rest} from 'tocco-app-extensions'
+import {consoleLogger} from 'tocco-util'
+
+import * as actions from './actions'
+
+export const inputSelector = state => state.input
+
+export default function* sagas() {
+  yield all([takeEvery(actions.FETCH_DATA, fetchData)])
+  yield all([takeEvery(actions.FETCH_FILECONTENT, fetchFileContent)])
+}
+
+export function* fetchData() {
+  const url = '/logfiles'
+  try {
+    const response = yield call(rest.simpleRequest, url)
+    let data = null
+
+    if (response) {
+      data = response.body.fileNames
+    }
+    yield put(actions.setData(data))
+  } catch (e) {
+    consoleLogger.logError('Failed to fetch data', e)
```

Comment:
suggestion(non-blocking): add notification toaster

if you have enough time you could show an error toaster here